lib/src/us/kbase/kbasefeaturevalues/KBaseFeatureValuesImpl.java

```diff
@@ -117,13 +119,20 @@ public void estimateK(EstimateKParams params,
                 params.getNeighbSize(), params.getMaxItems());
         provenance.get(0).withDescription("K estimation for K-Means clustering method")
                 .withInputWsObjects(Arrays.asList(params.getInputMatrix()));
-        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(params.getOutWorkspace())
+	WorkspaceClient wsc = getWsClient();
```

Comment:
This should probably be using DFU otherwise provenance won't be saved

Reply:
This applies to many places in this code - there's a ton of duplicated code here. DRYing it up would be good.

Reply:
The extra ws calls have been factored out. 

Since we are not testing provenance and no SDK utils for most objects will keep existing save methods.